scripts/travis/upload-to-docker.sh

```diff
@@ -22,23 +25,44 @@ echo "REPO=$REPO, BRANCH=$BRANCH, TAG=$TAG, IMAGE=$IMAGE"
 # add major, major.minor and major.minor.patch tags
 if [[ -n ${major:-} ]]; then
   docker tag $IMAGE $REPO:${major}
+  docker tag $IMAGE ${QUAY_PREFIX}/$REPO:${major}
   if [[ -n ${minor:-} ]]; then
     docker tag $IMAGE $REPO:${major}.${minor}
+    docker tag $IMAGE ${QUAY_PREFIX}/$REPO:${major}.${minor}
```

Comment:
This doesn't look quite right - this tagging will apply to Docker as well? Maybe we can simply prefix the REPO at the beginning once if quay login is in effect, and run the same set of commands otherwise. Let's separate the logic of the script from the destination registry.

Reply:
If I get it correctly, we should make the script upload-to-docker.sh to accept a parameter with acceptable values as "dockerhub" or "quay" and should remove all if dockerhub/quay conditions and make the script agnostic of the upstream registry ?

Reply:
this is what I am suggesting:

1. In the workflows, we call `upload-to-docker.sh` and pass secrets as env vars (instead of having a dedicated login action)
2. `upload-to-docker.sh` looks like this:

```sh
function do_upload() {
    registry = $1
    prefix = $2
    token_var = $3

    marker=.$registry.login
    if [ ! -f $marker && -v $token_var ]; then
        printenv $token_var | docker login ...
        touch marker
    fi

    if [ ! -f $marker ]; then
        echo "skipping upload to $registry, not logged in:
        return
    fi

    # current content of upload-to-docker.sh before this PR
}

do_upload "docker.io" "" "DOCKERHUB_TOKEN"
do_upload "quay.io" "quay prefix" "QUAY_TOKEN"
```

This way, the _only_ thing that individual actions need to do about logging in is pass the token as env. The login logic is encapsulated into the upload script, the fact that it does docker and quay is also encapsulated, and the login is efficient by using a cache "marker" file

Reply:
Sounds great Yuri.